#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.1485471937004881 -0.9808635788473236 -0.12585853540594655 1.5596251705548159
  position -3.3456313331999405 1.0317224459348158 1.688518265039086
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
DEF OMNI_WHEELS Robot {
  translation -1.491204262526917 0.06648242793370679 1.2000160770075072
  rotation -0.025376621068029678 0.9965422846115447 0.07911701513816481 0.5399701889189173
  children [
    DEF SOLID1 Solid {
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            position -52.47434076451028
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor1"
              maxVelocity 52.35
              maxTorque 0.5935
            }
            PositionSensor {
              name "encoder_1"
              resolution 0.00523597
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 -1 2.208858307073589
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position 31.239134691208243
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 6.106393462489897
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.02
                            radius 0.01
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -82.52585107904721
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 5.438743221466996
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position -307.19456592131
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation -1 0 0 5.601671176689865
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position -133.56291125956062
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 4.667165498390281
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position -106.09582743404701
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 -1 0 5.5648625191736265
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position -67.27788708969828
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 -1 0 4.446034017902417
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position 381.30934090956856
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865475 0.7071067811865475 0 4.3182224787933725
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position 43.75211263635815
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 6.05300079328063
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  DEF WHEEL_SHAPE Shape {
                    appearance PBRAppearance {
                      baseColor 0.678431 0.847059 0.901961
                      roughness 1
                      metalness 0
                    }
                    geometry Cylinder {
                      height 0.02
                      radius 0.049999999999999996
                      subdivision 20
                    }
                  }
                ]
              }
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 1"
      boundingObject DEF DUMMY Sphere {
        radius 0.01
      }
      physics Physics {
      }
    }
    DEF SOLID2 Solid {
      rotation 0 1 0 2.09
      children [
        DEF WHEEL2 HingeJoint {
          jointParameters HingeJointParameters {
            position 50.26224486307741
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor2"
              maxVelocity 52.35
              maxTorque 0.5931
            }
            PositionSensor {
              name "encoder_2"
              resolution 0.0052359
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 -0.9999999999999999 0.0008317843593314265
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position 40.85415052104539
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 3.155038677967872
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -78.39503423418935
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation -0.7071067811865475 -0.7071067811865475 0 2.9968105480343152
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position 13.574558162464475
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 1.0081875481053024
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position -280.4362869604025
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation -0.7071067811865475 -0.7071067811865475 0 3.9761334445007215
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position 60.54278224294966
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 -1 0 2.289070828846202
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position -279.90143400406413
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 -1 0 3.441280488162324
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position 11.578413594029556
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 -0.7071067811865476 0 0.9879570203296174
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position -352.84463405270697
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation 0.7071067811865475 -0.7071067811865475 0 0.9862568506501226
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 2"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF SOLID3 Solid {
      rotation 0 1 0 4.18
      children [
        DEF WHEEL3 HingeJoint {
          jointParameters HingeJointParameters {
            position -0.3824457684788737
            axis 0 0 1
            anchor 0 0 0.1
          }
          device [
            RotationalMotor {
              name "motor3"
              maxVelocity 52.35
              maxTorque 0.5931
            }
            PositionSensor {
              name "encoder_3"
              resolution 0.0052359
            }
          ]
          endPoint Solid {
            translation 0 0 0.1
            rotation 0 0 -1 0.37853632847930596
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  position -472.00247017620654
                  anchor 0 0.05 0.02
                }
                endPoint Solid {
                  translation 0 0.05 0.02
                  rotation 1 0 0 5.519613169442029
                  children [
                    USE r1
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  position -462.3445630179654
                  axis 0.707107 0.707107 0
                  anchor -0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 0.03535 0.02
                  rotation 0.7071067811865475 0.7071067811865475 0 2.6111497133239863
                  children [
                    USE r2
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  position -132.9836448730174
                  anchor 0 -0.05 0.02
                }
                endPoint Solid {
                  translation 0 -0.05 0.02
                  rotation 1 0 0 5.246431884933505
                  children [
                    USE r3
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  position -131.9529091762287
                  axis 0.707107 0.707107 0
                  anchor 0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 -0.03535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 6.277167581722172
                  children [
                    USE r4
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  position -374.21197789222646
                  axis 0 1 0
                  anchor -0.05 0 0.02
                }
                endPoint Solid {
                  translation -0.05 0 0.02
                  rotation 0 1 0 2.7791405385487313
                  children [
                    USE r5
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  position 63.684061997903754
                  axis 0 1 0
                  anchor 0.05 0 0.02
                }
                endPoint Solid {
                  translation 0.05 0 0.02
                  rotation 0 1 0 0.8522089261078896
                  children [
                    USE r6
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  position -1579.6832312230106
                  axis -0.707107 0.707107 0
                  anchor -0.03535 -0.03535 0.02
                }
                endPoint Solid {
                  translation -0.03535 -0.03535 0.02
                  rotation -0.7071067811865475 0.7071067811865475 0 3.6794661862452123
                  children [
                    USE r7
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  position 40.13507252271421
                  axis -0.707107 0.707107 0
                  anchor 0.03535 0.03535 0.02
                }
                endPoint Solid {
                  translation 0.03535 0.03535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 2.435960679636692
                  children [
                    USE r8
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              USE WHEEL_TRANS
            ]
            boundingObject Transform {
              translation 0 0 0.02
              rotation 1 0 0 1.5708
              children [
                DEF wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    transparency 0.12999999523162842
                    roughness 1
                    metalness 0
                    emissiveColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.02
                    radius 0.05
                  }
                }
              ]
            }
            physics Physics {
            }
          }
        }
      ]
      name "wheel 3"
      boundingObject USE DUMMY
      physics Physics {
      }
    }
    DEF BODY_TRANS Transform {
      translation 0 0.03 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.5647058823529412 0.9333333333333333 0.5647058823529412
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.15
            radius 0.1
            subdivision 18
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.07 0.05 0.07
      rotation 0 1 0 5.672386513999999
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.6431372549019608 0 0
          }
          geometry Box {
            size 0.02 0.01 0.01
          }
        }
      ]
      name "distance_sensor1"
      lookupTable [
        0.02 0 0
        5 1659 0
      ]
    }
    DistanceSensor {
      translation 0.1 0.05 0.02
      rotation 0 0.9999999999999999 0 6.108585901000003
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.6431372549019608 0 0
          }
          geometry Box {
            size 0.02 0.01 0.01
          }
        }
      ]
      name "distance_sensor2"
      lookupTable [
        0.02 0 0
        5 1659 0
      ]
    }
  ]
  boundingObject USE BODY_TRANS
  physics Physics {
  }
  controller "u3_exam_Chacon_Ivan"
}
Solid {
  translation 2.7610582444496283 0.3 3.0040890410718886
  rotation 0 -0.9999999999999999 0 4.974188401981909
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry DEF G_OBSATCLE Box {
        size 1 0.6 0.1
      }
    }
  ]
  name "solid(3)"
  boundingObject USE G_OBSATCLE
}
Solid {
  translation 0.6199999410032578 0.3 0.6000000669438965
  rotation 0 -1 0 1.5707963709819095
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry DEF G_OBSATCLE Box {
        size 0.9 0.6 0.1
      }
    }
  ]
  name "solid(2)"
  boundingObject USE G_OBSATCLE
}
